applications/MeshMovingApplication/python_scripts/ale_fluid_solver.py

```diff
@@ -107,25 +113,42 @@ def AddVariables(self):
             KM.Logger.PrintInfo("::[AleFluidSolver]::", "Variables Added")
 
     def AddDofs(self):
-        self.mesh_motion_solver.AddDofs()
+        self.mesh_motion_solver_full_mesh.AddDofs()
         self.fluid_solver.AddDofs()
         if self.is_printing_rank:
             KM.Logger.PrintInfo("::[AleFluidSolver]::", "DOFs Added")
 
     def Initialize(self):
         # Saving the ALE-interface-parts for later
         # this can only be done AFTER reading the ModelPart
-        self.ale_boundary_parts = []
         main_model_part_name = self.settings["fluid_solver_settings"]["model_part_name"].GetString()
 
         ale_boundary_parts_params = self.settings["ale_boundary_parts"]
-
+        self.ale_boundary_parts = []
         for i_name in range(ale_boundary_parts_params.size()):
             sub_model_part_name = ale_boundary_parts_params[i_name].GetString()
             full_model_part_name = main_model_part_name + "." + sub_model_part_name
             self.ale_boundary_parts.append(self.model[full_model_part_name])
 
-        self.mesh_motion_solver.Initialize()
+        mesh_motion_parts_params = self.settings["mesh_motion_parts"]
+        self.mesh_motion_solvers = []
+        if mesh_motion_parts_params.size() == 0:
+            # the entire Fluid-ModelPart is used in the Mesh-Solver
+            self.mesh_motion_solvers.append(self.mesh_motion_solver_full_mesh)
+        else:
+            # SubModelParts of the Fluid-ModelPart are used in the Mesh-Solver
+            # each SubModelPart has its own mesh-solver
+            for i_name in range(mesh_motion_parts_params.size()):
+                sub_model_part_name = mesh_motion_parts_params[i_name].GetString()
+                full_model_part_name = main_model_part_name + "." + sub_model_part_name
+                sub_mesh_solver_settings = self.settings["mesh_motion_solver_settings"].Clone()
+                sub_mesh_solver_settings["model_part_name"].SetString(full_model_part_name)
+
+                self.mesh_motion_solvers.append(mesh_mothion_solvers_wrapper.CreateSolverByParameters(
```

Comment:
`mesh_motion_solvers_wrapper`

Reply:
this name is used already, I don't understand

Reply:
typo : `mesh_mothion_solvers_wrapper` -> `mesh_motion_solvers_wrapper`

Reply:
I will do this in the next PR, since I don't want to wait again for the CI